lib/src/dynamics/island.dart

```diff
@@ -470,18 +470,20 @@ class Island {
   void addBody(Body body) {
     assert(_bodyCount < _bodyCapacity);
     body._islandIndex = _bodyCount;
-    _bodies[_bodyCount] = body;
-    ++_bodyCount;
+    _bodies.add(body);
+    _bodyCount++;
   }
 
   void addContact(Contact contact) {
     assert(_contactCount < _contactCapacity);
-    _contacts[_contactCount++] = contact;
+    _contacts.add(contact);
+    _contactCount++;
```

Comment:
do we even need _contactCount? could be _contacts.length or size on decent languages haha

Reply:
It is using arrays in a very C++ way, where things is related depending on what index they are on, and all indices not always used. I tried to refactor it away but everything crashed. Might do another try on it later now when I have removed all the pools.